src/meta/src/barrier/mod.rs

```diff
@@ -174,23 +183,57 @@ pub struct GlobalBarrierManager<S: MetaStore> {
 }
 
 struct CheckpointControl<S: MetaStore> {
-    /// Signal whether the barrier with building actor is sent
-    is_build_actor: bool,
     /// Save the state and message of barrier in order
     command_ctx_queue: VecDeque<EpochNode<S>>,
 }
+/// The changed table id vec.
+#[derive(Default)]
+pub struct ChangedTableId {
+    create_table_id: Vec<TableId>,
+    drop_table_id: Vec<TableId>,
+}
+impl ChangedTableId {
+    pub fn can_actor_send_or_collect(&self, s: ActorState, table_id: &TableId) -> bool {
+        s == ActorState::Running && !self.drop_table_id.contains(table_id)
+            || s == ActorState::Inactive && self.create_table_id.contains(table_id)
+    }
+
+    pub fn new<'a, S: MetaStore>(
+        queue_changed_table: impl Iterator<Item = &'a EpochNode<S>>,
+        new_changed_table: ChangedTableState,
+    ) -> Self {
+        let mut create_table_id = vec![];
+        let mut drop_table_id = vec![];
+        queue_changed_table.for_each(|node| match node.command_ctx.command.changed_table_id() {
+            Create(table_id) => create_table_id.push(table_id),
+            Drop(table_id) => drop_table_id.push(table_id),
+            _ => {}
+        });
```

Comment:
Is there any possibility that a mview is first created and then dropped? Do we cover this case?

Reply:
And If we want to drop mv. It must be finished creating. So there is no possibility that created and dropped can be in queue at the same time. So I think we can drop mv like normal.

Reply:
Cool. So how about adding a `debug_assert` here to check the table id not in the `create_table_ids` when pushing into `drop_table_ids`?